language_service/src/code_lens.rs

```diff
@@ -25,6 +25,38 @@ pub(crate) fn get_code_lenses(
         return vec![];
     }
 
+    // Don't show code lenses if there are compilation or project errors
+    // that would prevent program execution
+    if !compilation.project_errors.is_empty() {
+        return vec![];
+    }
+
+    // Don't show code lenses if there are compilation errors that would prevent execution
+    let has_blocking_errors = compilation.compile_errors.iter().any(|error| {
+        match error.error() {
+            // These error types prevent execution
+            qsc::compile::ErrorKind::Frontend(_)
+            | qsc::compile::ErrorKind::DependencyCycle
+            | qsc::compile::ErrorKind::CircuitParse(_)
+            | qsc::compile::ErrorKind::OpenQasm(_) => true,
+            // Pass errors may or may not prevent execution - check the specific error type
+            qsc::compile::ErrorKind::Pass(pass_error) => {
+                match pass_error {
+                    // Entry point errors don't prevent individual operation execution
+                    qsc_passes::Error::EntryPoint(_) => false,
+                    // Other pass errors prevent execution
+                    _ => true,
+                }
+            }
+            // Lint errors don't prevent execution
+            qsc::compile::ErrorKind::Lint(_) => false,
+        }
+    });
+
+    if has_blocking_errors {
+        return vec![];
+    }
+
```

Comment:
This code is largely unnecessary and is checking extra things that don't apply. This should be replaced with:

```suggestion
    if !compilation.project_errors.is_empty()
        || compilation
            .compile_errors
            .iter()
            .any(|e| !matches!(e.error(), ErrorKind::Lint(..)))
    {
        // Don't show code lenses if there are any project errors or non-Lint errors in the compilation.
        return vec![];
    }
```

Reply:
@swernli why? Entrypoint errors don't block any of the code lens commands, so excluding them made sense. (It's also nice to be able to just write an operation into a file and run it, without worrying about naming it `Main`)

Reply:
WHAT?! This changes everything.